testdata/btf_map_init.c

```diff
@@ -0,0 +1,33 @@
+#include "common.h"
+
+#if __clang_major__ >= 9
+
+int __section("socket/1") tail_1() {
+	return 42;
+}
+
+// Tail call map (program array) initialized with program pointers.
+struct {
+	__uint(type, BPF_MAP_TYPE_PROG_ARRAY);
+	__uint(key_size, sizeof(uint32_t));
+	__uint(max_entries, 2);
+	__array(values, int ());
```

Comment:
this is nifty, I didn't realize `int()` can be used with typeof() to get a function prototype type recorded. Nice!